lib/ramble/ramble/test/end_to_end/spack_env_cache.py

```diff
@@ -70,50 +74,53 @@ def test_spack_env_cache():
-    with open(test1_log, "r") as f:
-        content = f.read()
-        assert "spack install" in content
-        assert "spack concretize" in content
-
-    test2_log = os.path.join(ws.log_dir, "setup.latest", "gromacs.water_bare.test2.out")
-    with open(test2_log, "r") as f:
-        content = f.read()
-        assert "spack install" in content
-        assert "spack concretize" in content
-
-    # Envs should already exist and can skip spack calls.
-    test3_log = os.path.join(ws.log_dir, "setup.latest", "gromacs.water_bare.test3.out")
-    with open(test3_log, "r") as f:
-        content = f.read()
-        assert "spack install" not in content
-        assert "spack concretize" not in content
-
-    test4_log = os.path.join(ws.log_dir, "setup.latest", "gromacs.water_gmx50.test4.out")
-    with open(test4_log, "r") as f:
-        content = f.read()
-        assert "spack install" not in content
-        assert "spack concretize" not in content
+    try:
+        workspace_name = "test-spack-env-cache"
+        ws = ramble.workspace.create(workspace_name)
+        ws.write()
+
+        config_path = os.path.join(ws.config_dir, ramble.workspace.config_file_name)
+
+        with open(config_path, "w+") as f:
+            f.write(test_config)
+
+        ws._re_read()
+
+        workspace(
+            "setup",
+            "--dry-run",
+            global_args=["-w", workspace_name],
+        )
+
+        # spack env should be present only at the env_name level.
+        assert os.path.exists(os.path.join(ws.software_dir, "gromacs"))
+        assert os.path.exists(os.path.join(ws.software_dir, "g2"))
+        assert not os.path.exists(os.path.join(ws.software_dir, "g2.water_bare"))
+
+        # First encounter of an env_name (test1 -> gromacs, test2 -> g2) requires spack usage.
+        test1_log = os.path.join(ws.log_dir, "setup.latest", "gromacs.water_bare.test1.out")
+        with open(test1_log, "r") as f:
+            content = f.read()
+            assert "spack install" in content
+            assert "spack concretize" in content
+
+        test2_log = os.path.join(ws.log_dir, "setup.latest", "gromacs.water_bare.test2.out")
+        with open(test2_log, "r") as f:
+            content = f.read()
+            assert "spack install" in content
+            assert "spack concretize" in content
+
+        # Envs should already exist and can skip spack calls.
+        test3_log = os.path.join(ws.log_dir, "setup.latest", "gromacs.water_bare.test3.out")
+        with open(test3_log, "r") as f:
+            content = f.read()
+            assert "spack install" not in content
+            assert "spack concretize" not in content
+
+        test4_log = os.path.join(ws.log_dir, "setup.latest", "gromacs.water_gmx50.test4.out")
+        with open(test4_log, "r") as f:
+            content = f.read()
+            assert "spack install" not in content
+            assert "spack concretize" not in content
+    except RunnerError as e:
```

Comment:
Curious why this is needed now?

Reply:
It is not strictly needed in this PR, I just noticed it when I was going through the unit tests.

I believe if you take develop today on a machine that doesn't have spack installed, this unit test will fail. This change allows it to be skipped instead.